expression/builtin_time_vec.go

```diff
@@ -935,11 +935,41 @@ func (b *builtinWeekWithModeSig) vecEvalInt(input *chunk.Chunk, result *chunk.Co
 }
 
 func (b *builtinExtractDurationSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinExtractDurationSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	unit, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(unit)
+	if err := b.args[0].VecEvalString(b.ctx, input, unit); err != nil {
+		return err
+	}
+	dur, err := b.bufAllocator.get(types.ETDuration, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(dur)
+	if err = b.args[1].VecEvalDuration(b.ctx, input, dur); err != nil {
+		return err
+	}
+	result.ResizeInt64(n, false)
+	result.MergeNulls(unit, dur)
+	i64s := result.Int64s()
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		durI, unitI := dur.GetDuration(i, int(types.UnspecifiedFsp)), unit.GetString(i)
```

Comment:
`Duration` is a fixed-length type, consider using https://godoc.org/github.com/pingcap/tidb/util/chunk#Column.GoDurations

Reply:
@icditwang Friendly ping, please address this comment.

Reply:
ok...I forgot it.